src/Sentry/Internal/MainSentryEventProcessor.cs

```diff
@@ -108,9 +113,11 @@ public SentryEvent Process(SentryEvent @event)
             }
 
             // Report local user if opt-in PII, no user was already set to event and feature not opted-out:
-            if (_options.SendDefaultPii && _options.IsEnvironmentUser && !@event.HasUser())
+            if (_options.SendDefaultPii && _options.IsEnvironmentUser)
```

Comment:
`IsEnvironmentUser` this option control whether to use `Environment.UserName` or not. So I wouldn't guard your new option under this condition:


```csharp
if (_options.SendDefaultPii)
{
    if (_options.IsEnvironmentUser && !@event.HasUser())
    {
        @event.User.Username = Environment.UserName;
    }
    @event.User.IpAddress ??= UserIpServerInferred;
}

```

Reply:
ty for pointing out.